src/main/java/io/vertx/core/http/impl/Http2HeadersAdaptor.java

```diff
@@ -23,12 +25,18 @@
 import java.util.Set;
 import java.util.stream.Collectors;
 
+import static io.netty.util.AsciiString.*;
+
 /**
  * @author <a href="mailto:nmaurer@redhat.com">Norman Maurer</a>
  */
 public class Http2HeadersAdaptor implements MultiMap {
 
   static CharSequence toLowerCase(CharSequence s) {
+    if (s instanceof AsciiString) {
```

Comment:
how is this related to this PR ?

Reply:
It's not strictly related to this PR. But while working on this (and thinking about unnecessary conversions) I stumbled upon this method.

By the way, this property of AsciiString is important to note, if we want to resurrect #1878 : having the HttpHeaders constant all lower case avoids repeated conversion for HTTP2